### ISSUE TITLE
`Duplicate array keys` delete automatically

### ISSUE THREAD

**Issue by vladyslavstartsev** [kalessil/phpinspectionsea#528]
I was editing translation files in `Yii2`/`Laravel` which use `array` for it. Sometimes if the array is huge, it will have no only duplicated keys but also values.
For example. ([full array](https://github.com/adam2314/linet3/blob/master/protected/messages/he_il/app.php))
```php
return [
    'Profit & loss' => 'רווח הפסד',
    'Proforma' => 'חשבונית עסקה',
    'Quote' => 'הצעת מחיר',
    'Receipt' => 'קבלה',
    'Reconciliations' => 'התאמות',
    'Register' => 'קופה',
    'Proforma' => 'חשבונית עסקה',
    'Reports' => 'דוחות',
];
```

the `Proforma` key is the same in this example.
I mean, in high level overview it can be something like
1. If duplicated keys was found then
2. Check if it can be delete automatically (what I mean is)
2.1. Check if it's (value) static (like example above, usually it's some translation files) (seems easy one)
2.2. Or (if non static values) check if it calls the same function/method
3. Suggest to delete `Duplicate array keys` if values is the same. If not (value static but not the same) write something like `key duplicated but value is not the same`

**Comment by kalessil** [comment]
So we need a more intelligent version of a standard PS inspection (PhpDuplicateArrayKeysInspection)?

**Comment by vladyslavstartsev** [comment]
@kalessil kind of yes

**Comment by kalessil** [comment]
Good, then let's make a new one =)

**Comment by kalessil** [comment]
Re-scheduling for 2.3+, 2.3.11 is full.

**Comment by kalessil** [comment]
Re-opening

**Comment by kalessil** [comment]
Bad bot, I've wrote "reopening".

**Comment by rentalhost** [comment]
Maybe reconfigure it to 2 years instead of one?

**Comment by kalessil** [comment]
Most probably I'll do it =)

**Comment by kalessil** [comment]
Implemented!
